### DISCUSSION TITLE
Where should the types be put?

### DISCUSSION THREAD

**Discussion by BlueskyFR** [nuxt/nuxt#29940] [Questions]
Hey!
When wanting to share types between the server and the client, where should I defined my types and/or my schemas?
Is there something in the lines of a `types/` directory that could be auto-imported or something?

I found absolutely NOTHING about that in the docs, but it seems so obvious since we are using Typescript!

So maybe I got something wrong and we are not supposed to define types separately?

I'd love an official answer to that question.
Thanks in advance for the answers!

**Comment by darthf1** [comment]
https://nuxt.com/blog/v3-14#shared-folder-for-code-and-types-shared-with-clientserver

**Reply by BlueskyFR** [reply]
Thanks for that!!
Wow this is very new then.
The problem is that the directory structure in the docs does not refer that at all??

**Reply by BlueskyFR** [reply]
Also whenever I put something in the shared folder it seems that autoimports do not work!
Should I do something specific to enable them? VSCode automatically adds the import statement when I autocomplete a type placed in `shared/types/mytypes.d.ts` as `export type myType = string;`.
